sklearn/tests/test_docstring_parameters.py

```diff
@@ -226,7 +227,11 @@ def test_fit_docstring_attributes(name, Estimator):
         # provided, this checks if the word "only" is present in the attribute
         # description, and if not the attribute is required to be present.
         if 'only ' not in desc:
-            assert hasattr(est, attr.name)
+            try:
+                assert hasattr(est, attr.name)
+            except FutureWarning:
+                # FutureWarning should be raised for deprecated attributes
+                assert_warns(FutureWarning, hasattr, est, attr.name)
```

Comment:
I would ignore the warnings with a comment on deprecated attributes. Since the warning was caught, the assert will always be true.

```py
with ignore_warnings():
	assert hasattr(est, attr.name)
```

Reply:
Tricky and easy solution. Thank you @thomasjpfan!